src/XML/src/FMI3/fmi3_xml_variable.c

```diff
@@ -1254,6 +1270,47 @@ static int fmi3_xml_variable_process_attr_multipleset(fmi3_xml_parser_context_t*
     return 0;
 }
 
+static int fmi3_xml_variable_process_attr_intermediateupdate(fmi3_xml_parser_context_t* context,
+        fmi3_xml_variable_t* variable, fmi3_xml_elm_enu_t elm_id)
+{
+    unsigned int intermediateUpdate;
+    fmi3_xml_model_description_t* md = context->modelDescription;
+
+    // peek due to "must not have attribute" error check
+    if (!fmi3_xml_peek_attr_str(context, fmi_attr_id_intermediateUpdate)) { // if attribute not set, use default
+        variable->intermediateUpdate = 0; // default
+        return 0;
+    } 
+    // else: attribute exists
+
+    fmi3_fmu_kind_enu_t fmuKind = fmi3_xml_get_fmu_kind(md);
+    // Spec: "This attribute is ignored in Model Exchange and Scheduled Execution", ignored unless Co-Simulation
```

Comment:
If ignored, should we still set the default value?
This attribute is only relevant for Co-Simulation.

Reply:
I don't think you should check for ignore before peeking. Won't you now give this Info message on all variables even when the attribute is not defined?

Reply:
Good catch, fixed and adjusted the order.